README.md

```diff
@@ -1,6 +1,79 @@
 # CNS Object Broker
-## For dynmically provisioning object storage compliant with an S3 interface.
+## Utilize Kubernetes Service-Catalog to dynamically provision CNS Object Storage.
 
 ## WARNING!
+The work in this project is a proof of concept and not intended for anything
+more than that.
 
-The work in this project is a proof of concept and not intended for anything more than that.
+## Overview
+A core feature of the Kubernetes system is the ability to provision a diverse
+offering of block and file storage on demand.  This project seeks to demonstrate
+that by using the Kubernetes-Incubator's Service-Catalog, it is now also possible
+to bring this dynamic provisioning to S3 object storage.  This broker is designed
+to be used with Gluster-Kubernetes S3.
+
+See our detailed [command flow diagram](./docs/Service-Catalog-Published.png) for
+ an overview of the system.
+
+### What It Does
+This broker implements OpenServiceBroker methods for creating, connecting to, and destroying
+object buckets.
+- For each new Service Instance, a new, unquiely named bucket is created.
```

Comment:
Do you want to mention that a bucket name can be defined as a parameter to the ServiceInstance? This handle the case of reading existing s3 data.

Reply:
Alternatively, what about removing the custom name option?  It really can be abstracted from the user.  To share access to the bucket to multiple pods, a user only needs to create a second service instance credential.

Reply:
I think a random name is difficult to share with others. Even a name with a fixed prefix and random characters as a suffix is difficult to use except for the single user use case. The ability to define the entire bucket name is important IMO. (I assume the name is name-spaced.) There could be corporate naming standards or conventions, etc. The bucket could be expected to persist and be managed for a long time, etc. What is the reason for not wanting the user to be able to specify their bucket name?

Reply:
The chief reason is that users won't use the name too connect to the bucket.  They only need to know the name of the service instance.  They would create a service instance credential to get the coordinates of the bucket.